pic-sure-auth-services/src/main/java/edu/harvard/hms/dbmi/avillach/auth/service/auth/FENCEAuthenticationService.java

```diff
@@ -159,108 +176,173 @@ public Response getFENCEProfile(String callback_url, Map<String, String> authReq
 	            ObjectMapper mapper = new ObjectMapper();
 	            // `JsonNode` to JSON string
 	            String prettyString = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(fence_user_profile);
-	            
-	            logger.trace("getFENCEProfile() user profile structure:"+  prettyString);
+                logger.trace("getFENCEProfile() user profile structure:{}", prettyString);
             }
-            logger.debug("getFENCEProfile() .username:" + fence_user_profile.get("username"));
-            logger.debug("getFENCEProfile() .user_id:" + fence_user_profile.get("user_id"));
-            logger.debug("getFENCEProfile() .email:" + fence_user_profile.get("email"));
+            logger.debug("getFENCEProfile() .username:{}", fence_user_profile.get("username"));
+            logger.debug("getFENCEProfile() .user_id:{}", fence_user_profile.get("user_id"));
+            logger.debug("getFENCEProfile() .email:{}", fence_user_profile.get("email"));
         } catch (Exception ex) {
-            logger.error("getFENCEProfile() could not retrieve the user profile from the auth provider, because "+ex.getMessage(), ex);
+            logger.error("getFENCEProfile() could not retrieve the user profile from the auth provider, because {}", ex.getMessage(), ex);
             throw new NotAuthorizedException("Could not get the user profile "+
                     "from the Gen3 authentication provider."+ex.getMessage());
         }
 
-        User current_user = null;
+        User current_user;
         try {
             // Create or retrieve the user profile from our database, based on the the key
             // in the Gen3/FENCE profile
             current_user = createUserFromFENCEProfile(fence_user_profile);
-            logger.info("getFENCEProfile() saved details for user with e-mail:"
-                    +current_user.getEmail()
-                    +" and subject:"
-                    +current_user.getSubject());
-            
+            logger.info("getFENCEProfile() saved details for user with e-mail:{} and subject:{}", current_user.getEmail(), current_user.getSubject());
+
             //clear some cache entries if we register a new login
             AuthorizationService.clearCache(current_user);
             UserService.clearCache(current_user);
-
         } catch (Exception ex) {
-            logger.error("getFENCEToken() Could not persist the user information, because "+ex.getMessage());
+            logger.error("getFENCEToken() Could not persist the user information, because {}", ex.getMessage());
             throw new NotAuthorizedException("The user details could not be persisted. Please contact the administrator.");
         }
 
-        // Update the user's roles (or create them if none exists)
-        //Set<Role> actual_user_roles = u.getRoles();
         Iterator<String> project_access_names = fence_user_profile.get("authz").fieldNames();
+
+        // Convert Iterator to a set of strings
```

Comment:
what? You're not converting anything in that LOC. What does this refer to?

Reply:
Oh I see, it's the whole block. This should probably be pulled out into a method